Add back Junit XML generation

edeati opened the pull request:
### Definition of Done

* [ ] Code has been peer reviewed
* [x] Test coverage has been maintained or improved
* [x] All tests pass within CI
* [x] README is up to date

edeati committed:
Add back Junit XML generation

edeati committed:
Forgot pom.xml from previous commit

edeati committed:
Initial checkin of AMTv4 changes

edeati committed:
Speed up the V4 flat file generation

edeati committed:
Change plugin version

edeati committed:
Add Deployment details

edeati committed:
Change java to 17

edeati committed:
Change java version for compiler

edeati committed:
Change Java to 11

edeati committed:
Add application/java-archive to the allowed zip file input

edeati committed:
AMT V3 backward compatible now

edeati committed:
Re-add tests

edeati committed:
Fix content test

edeati committed:
Remove extra stuff not needed

edeati committed:
Create parent directory for junit file if it doesn't exist

edeati committed:
My, oh my. You cannot delete artefacts from Azure Devops repo. If you delete it you cannot deplpoy a new one anymore. Just sad

edeati committed:
AMT V4 TP is not a parent of TPUU but a HAS_PRODUCT_NAME - test errors are now reporting an extra fail because I'm now logging no TP for a TPUU as a junit test fail

dionmcm committed:
updated allowed file size

dionmcm committed:
updated allowed file size

dionmcm committed:
need to cache more now that the AMT module is gone

dionmcm commented on azure-pipelines.yml:
Do we really want this file in the open source repository? I think maybe not.

Perhaps the open source repository should have GitHub Actions and push to Maven Central because it is open source software?

dionmcm commented on pom.xml:
```diff
@@ -151,14 +179,48 @@
 	</repositories>
+	<pluginRepositories>
+		<pluginRepository>
+			<id>${deployment.repo.id}</id>
```
Again should these repo variables be in the the POM in the open source version of this? Maybe they should and the CI (whatever we use like GitHub Actions) could populate these?

dionmcm commented on src/main/java/au/gov/digitalhealth/terminology/amtflatfile/Amt2FlatFile.java:
```diff
@@ -265,54 +272,57 @@ private void writeFlatFile(Path path) throws IOException {
                 Set<Concept> addedMpuus = new HashSet<>();
                 for (Concept tpuu : tpuus) {
-                    Concept tpuuTp = getParent(AmtConcept.TP, AmtConcept.TPUU, tpuu);
+                    Concept tpuuTp = conceptCache.isAmtV3() ? getParent(AmtRefset.TP, tpuu) : null;
```
This looks like it won't ever get a TPUU TP for AMT v3 anymore, but it probably should?

Should we make this branch AMT v4 only, or should we keep it supporting v3 or v4 whichever it is supplied?

dionmcm commented on src/main/java/au/gov/digitalhealth/terminology/amtflatfile/Amt2FlatFile.java:
```diff
@@ -352,12 +362,29 @@ private void writeFlatFile(Path path) throws IOException {
+                    testSuite.addTestCase("Mismatch", message, this.getClass().getName(), "MPP mismatch (" + mpp.getId() + ")", "ERROR");
+                }
+
+                if ( count % 10000 == 0) {
```
So that there's more logging as it goes
```suggestion
                if ( count % 1000 == 0) {
```

dionmcm commented on src/main/java/au/gov/digitalhealth/terminology/amtflatfile/Amt2FlatFile.java:
```diff
@@ -352,12 +362,29 @@ private void writeFlatFile(Path path) throws IOException {
+
+                if ( count % 10000 == 0) {
+                    long endTime = System.currentTimeMillis();
+                    logger.info("Processed CTPP [" + count + " of " + ctppCount + "] Time spent processing 10000 CTPPs is " + (endTime - startTime) + "ms");
```
```suggestion
                    logger.info("Processed CTPP [" + count + " of " + ctppCount + "] Time spent processing 1000 CTPPs is " + (endTime - startTime) + "ms");
```

dionmcm commented on src/main/java/au/gov/digitalhealth/terminology/amtflatfile/Amt2FlatFile.java:
```diff
@@ -377,35 +404,34 @@ private void writeReplacementsFile(Path path) throws IOException {
+        leafParents.addAll(concept.getAncestors().stream().collect(Collectors.toSet()));
+        leafParents = memberOf(leafParents, parentType);
 
         Set<Concept> redunantAncestors =
```
This used to remove redundant ancestors but doesn't anymore - is that just an error?

dionmcm commented on src/main/java/au/gov/digitalhealth/terminology/amtflatfile/AmtCache.java:
```diff
@@ -82,36 +126,43 @@ private void processAmtFiles(FileSystem amtZip) throws IOException {
             readFile(historicalFile, s -> handleHistoricalAssociationRefsetRow(s), true, "\t");
         }
 
+        if (this.isAmtV3()) {
```
Is it worth trying to refactor this code to only work with V4 or only V3? I.e. different versions of the flat file generator and we can cut down some code? Not sure how much it would change, maybe not with the effort?

dionmcm reviewed (commented):
Just one issue in this list which looks like it might be a defect? @lukeswindale it might be good for you to take a look at this too.

dionmcm committed:
updated to account for case change in en-AU to en-au matching SNOMED International file naming

dionmcm committed:
Add handling for active ingredients in Concept class

A new field 'ingredients' added to Concept class and 'addIngredient' method introduced to handle the Concept's ingredients. The error messages for failure scenarios during file processing in AmtCache class have also been enhanced to display the full row information. Besides, various scenarios are handled for MES Product (MP) parents based on their ingredients and exceptions are thrown accordingly along with logging warnings in Amt2FlatFile class. New active ingredient types added to the AttributeType class.

edeati committed:
Works on experimental files too

edeati committed:
Add TSV file generation and fix tests

edeati committed:
Throw exception if there are duplicated testcases in the file

edeati committed:
Fix logging

dionmcm committed:
Update src/main/java/au/gov/digitalhealth/terminology/amtflatfile/Amt2FlatFile.java

dionmcm committed:
Update src/main/java/au/gov/digitalhealth/terminology/amtflatfile/Amt2FlatFile.java
